pkg/agent/billing/billing.go

```diff
@@ -130,7 +161,46 @@ func RunBillingMetricsCollector(
 	}
 }
 
-func (s *metricsState) collect(logger *zap.Logger, store VMStoreForNode, metrics PromMetrics) {
+func collectMetricsForVM(ctx context.Context, logger *zap.Logger, vm *vmapi.VirtualMachine, metricsChan chan vmMetricsKV, promMetrics *PromMetrics) {
+	byteCounts, err := vm.GetNetworkUsage(ctx, 500*time.Millisecond)
+	if err != nil {
+		logger.Error("Failed to collect network usage", util.VMNameFields(vm), zap.Error(err))
+
+		var rootErr string
+		//nolint:errorlint // This is ok because vmapi guarantees the error types
+		switch err.(type) {
+		case vmapi.NetworkUsageEndpointNotFoundError:
+			rootErr = "Endpoint not found"
+		case vmapi.NetworkUsageTimeoutError:
+			rootErr = "Timeout"
```

Comment:
I'd probably use the same categories of errors as billing sending uses - JSON (although in this case it's unmarshaling, not marshaling) and HTTP code.

The intent behind those is that _generally_ other errors are pretty easy to make sense of if you only look at the root, but HTTP status code / JSON unmarshaling are hard to make sense of from the metrics.

Reply:
Switched it